psi4/CMakeLists.txt

```diff
@@ -161,6 +161,24 @@ else()
     message(STATUS "Disabled simint")
 endif()
 
+if(${ENABLE_BrianQC})
+    if(NOT DEFINED BrianQC_DIR)
+        message(FATAL_ERROR "BrianQC_DIR must be set to the root of a BrianQC install (BrianQC_DEV=OFF) or the root of a BrianQC development repo (BrianQC_DEV=ON) when using BrianQC")
+    endif()
+    if(${BrianQC_DEV})
+        message(STATUS "${Cyan}Using development BrianQC${ColourReset}: ${BrianQC_DIR}")
+        include_directories("${BrianQC_DIR}/src/brian_module/api/")
+        include_directories("${BrianQC_DIR}/src/brian_module/static_wrapper/")
+        link_directories("${BrianQC_DIR}/build/lib/")
+    else()
+        message(STATUS "${Cyan}Using installed BrianQC${ColourReset}: ${BrianQC_DIR}")
+        include_directories("${BrianQC_DIR}/include/")
+        link_directories("${BrianQC_DIR}/lib/")
+    endif()
+else()
+    message(STATUS "Disabled BrianQC")
+endif()
```

Comment:
Is CMake BrianQC's build system (or one of them)? Is it exporting a `BrianQCConfig.cmake` file? We like that setup because it can export a "target" with link and include properties attached. This keeps the info about how to use a project near the people who know how to use it (you) :-)

Even though as a private project you're a little different, I think for pattern's sake, it'd still be good for you to have a file like https://github.com/psi4/psi4/blob/master/external/upstream/gau2grid/CMakeLists.txt . Apart from `gau2grid` -> `BrianQC`, I think the main change is that the middle section https://github.com/psi4/psi4/blob/master/external/upstream/gau2grid/CMakeLists.txt#L12-L30 becomes just a `message` command to contact you and arrange for a license.

Let me know if I can point you to code models.

Reply:
Thanks for the suggestion! As you wrote, since Brian is closed-source, there are a few differences, such as Psi4 not being able to automatically build us. Also, since Brian is not normally installed from source or a package manager, the location on the machine is much less predictable, so writing a FindBrianQC.cmake (which, if I understood correctly, is required by Psi4's 3rdparty dependecy cmakes, such as the gau2grid one) might be difficult. That's why originally we opted for letting the user specify everything explicitly. But you are right that we should streamline the build process somehow, so we are going to look into this.

Reply:
The nice thing about the `<project>Config.cmake` + `find_package(<project> CONFIG)` setup is that it works with package managers, from source, or in-between since it's separating the internal install layout (known in project's build system) from the location of that layout (needed by psi's build/detect system). Here's some links, in case helpful. 

* https://cmake.org/cmake/help/latest/manual/cmake-packages.7.html#creating-relocatable-packages
* https://gitlab.com/libxc/libxc/-/blob/master/CMakeLists.txt#L485-506
* https://github.com/dgasmith/gau2grid/blob/master/cmake/gau2gridConfig.cmake.in

Reply:
I have created a BrianQCConfig.cmake, and I have rewritten the Psi4-BrianQC linking to more closely match the conventions. Please let me know if there is anything else I should change.

Reply:
You are right, the global include_directories is not necessary. I've moved the includes into the target-specific conditional blocks.